src/mscorlib/src/Microsoft/Win32/Win32Native.cs

```diff
@@ -948,17 +948,45 @@ internal static bool DoesWin32MethodExist(String moduleName, String methodName)
                                 dwFlagsAndAttributes, hTemplateFile );
 
             return handle;
-        }            
-    
+        }
+
         // Do not use these directly, use the safe or unsafe versions above.
         // The safe version does not support devices (aka if will only open
         // files on disk), while the unsafe version give you the full semantic
         // of the native version.
-        [DllImport(KERNEL32, SetLastError=true, CharSet=CharSet.Auto, BestFitMapping=false)]
-        private static extern SafeFileHandle CreateFile(String lpFileName,
-                    int dwDesiredAccess, System.IO.FileShare dwShareMode,
-                    SECURITY_ATTRIBUTES securityAttrs, System.IO.FileMode dwCreationDisposition,
-                    int dwFlagsAndAttributes, IntPtr hTemplateFile);
+        [System.Security.SecurityCritical]
+        private static SafeFileHandle CreateFile(
+            string lpFileName,
+            int dwDesiredAccess,
+            System.IO.FileShare dwShareMode,
+            SECURITY_ATTRIBUTES securityAttrs,
+            System.IO.FileMode dwCreationDisposition,
+            int dwFlagsAndAttributes,
+            IntPtr hTemplateFile)
+        {
+#if PLATFORM_UNIX && FEATURE_CORECLR
+            return System.IO.FileStream.OpenHandle(lpFileName, dwCreationDisposition, (System.IO.FileAccess)dwDesiredAccess, (System.IO.FileOptions)dwFlagsAndAttributes);
```

Comment:
It won't work well if the SafeFileHandle happened to be passed to some other CoreCLR PAL method, e.g. CreateFileMapping... .